java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java

```diff
@@ -147,6 +152,23 @@ public BarrageTable entireTable() throws InterruptedException {
 
     @Override
     public synchronized BarrageTable partialTable(RowSet viewport, BitSet columns) throws InterruptedException {
+        // notify user when connection has already been used and closed
+        if (prevUsed) {
+            throw new UnsupportedOperationException("Cannot create multiple snapshots on a single connection");
```

Comment:
"connection" feels out of place here. Not worth delaying merge for, though.
```suggestion
            throw new UnsupportedOperationException("Snapshot object already used");
```